### DISCUSSION TITLE
Disabled submit button

### DISCUSSION THREAD

**Discussion by palomamtnez** [solspace/craft-freeform#119] [Questions & Answers] [answered]
I have a project that uses a freeform contact form.
It contains basic fields such as name, email, message and the submit button.

For some reason, after I upgraded the plugin from 3.8.4 to 3.10.10, the submit button is always disabled.
As a quick solution, I have taken a look at `vendor/solspace/craft-freeform/packages/plugin/src/Resources/js/scripts/front-end/plugin/freeform.js`
and in the section of `this.lockSubmit`, I saw that there's this line:
`(o||e)&&(s.disabled=!0)`
the first variable, `o`, returns `false`, but the variable `e` returns `true`.
Not sure what I am doing wrong that this variable is returning true and as a consequence, disabling the submit button.

Has anyone faced this issue before? 
FYI, I am using Vue 2, and as soon as I disable Vue, the submit button is not disabled anymore and the form works fine!

Any help is appreciated!

Freeform version I am currently using: 3.10.10 (again, it worked fine with the version 3.8.4)
Craft version: 3.6.10
If you need anything else I can give you more info

Thanks

**Comment by kevinamezaga** [comment]
Hi @palomamtnez, do you mean the submit button is disabled on submit or disabled in general?

Disabled on Submit is in general settings, but otherwise could it be a CSS class that's added by the Vue formatting template on Freeform?

<img width="841" alt="Screen Shot 2021-03-18 at 11 38 22" src="https://user-images.githubusercontent.com/25387497/111653886-a1161300-87de-11eb-984d-a57deafd5986.png">

**Reply by palomamtnez** [reply]
Hi @kevinamezaga 
Yip, I mean that the button is disabled in general. 

I have that lightbox field set to NO; I tried changing it to yes as well to see what happens, but it doesn't make any difference. 
As soon as the page loads, the button has the attribute `disabled="disabled"`, which makes it impossible to submit the form.

I tried to hardcode it just by adding the code:
`<button  type="submit" name="form_page_submit" class="button primary">Submit</button>`

But then it loads it and adds the disabled attribute again.

The template for the freeform form is on a twig file... but Vue is still loaded in the project as it is needed for a search bar on the to nav bar I have on the site.

Not sure what could be wrong ?

**Reply by kevinamezaga** [reply]
@palomamtnez, are you using conditionals?

**Comment by kjmartens** [comment]
I'm not sure exactly what the issue is here, but please note that in Freeform 3.10, we made some large changes and improvements to the Freeform javascript "plugin" (the JS that Freeform inserts into pages that contain Freeform forms). While everything you did to override some Freeform JS functionality (if applicable) should be backward compatible, etc, it's very possible some of your other scripts _could_ conflict in Freeform 3.10+ and need to be adjusted to account for it.

But to clarify, you have this setting disabled inside the Freeform settings?
<img width="395" alt="Screen Shot 2021-03-22 at 9 42 26 AM" src="https://user-images.githubusercontent.com/1821674/112007859-032e8b00-8af3-11eb-8cb2-40fc1babc8ed.png">

I'll run this by our lead developer though and see if he has any ideas about your suspicion with the `freeform.js` file. :)

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi @palomamtnez, this is caused by the Freeform plugin init. It locks the submit buttons until the document is loaded and all of the additional listeners and handlers are attached to the fields and then it unlocks the submit buttons.

I'm guessing there's no way around this if you're using Vue, so I will remove this functionality since it seems to do more harm than good.

**Comment by kjmartens** [comment]
@palomamtnez this should be ready now in Freeform 3.10.11 :)
